utils/wasm-gen/src/config/syscalls/injection.rs

```diff
@@ -0,0 +1,121 @@
+
+use crate::InvocableSysCall;
+
+use gear_wasm_instrument::syscalls::SysCallName;
+use std::{collections::HashMap, ops::RangeInclusive};
+
+/// This enum defines how the sys-call should be injected into wasm module.
+#[derive(Debug, Clone)]
+pub enum SysCallInjectionType {
+    /// Don't modify wasm module at all.
+    None,
+    /// Sys-call import will be injected into import section of wasm module,
+    /// but the `wasm-gen` generators will not call that sys-call.
+    Import,
+    /// Sys-call import will be injected into import section of wasm module,
+    /// and the `wasm-gen` generators will insert invoke instructions for that sys-call.
+    Function(RangeInclusive<u32>),
+}
+
+/// Possible injection types for each sys-call.
+#[derive(Debug, Clone)]
+pub struct SysCallsInjectionTypes(HashMap<InvocableSysCall, SysCallInjectionType>);
+
+impl SysCallsInjectionTypes {
+    /// Instantiate a sys-calls map, where each gear sys-call is injected into wasm-module only once.
+    pub fn all_once() -> Self {
+        Self::new_with_injection_type(SysCallInjectionType::Function(1..=1))
+    }
+
+    /// Instantiate a sys-calls map, where no gear sys-call is ever injected into wasm-module.
+    pub fn all_never() -> Self {
+        Self::new_with_injection_type(SysCallInjectionType::None)
+    }
+
+    /// Instantiate a sys-calls map with given injection type.
+    fn new_with_injection_type(injection_type: SysCallInjectionType) -> Self {
+        let sys_calls = SysCallName::instrumentable();
+        Self(
+            sys_calls
+                .iter()
+                .cloned()
+                .map(|name| (InvocableSysCall::Loose(name), injection_type.clone()))
+                .chain(sys_calls.iter().cloned().filter_map(|name| {
+                    InvocableSysCall::has_precise_variant(name)
+                        .then_some((InvocableSysCall::Precise(name), injection_type.clone()))
+                }))
+                .collect(),
+        )
+    }
+
+    /// Gets injection type for given sys-call.
+    pub fn get(&self, name: InvocableSysCall) -> SysCallInjectionType {
+        self.0
+            .get(&name)
+            .cloned()
+            .expect("instantiated with all sys-calls set")
+    }
+
+    /// Sets possible amount range for the the sys-call.
+    pub fn set(&mut self, name: InvocableSysCall, min: u32, max: u32) {
+        self.0
+            .insert(name, SysCallInjectionType::Function(min..=max));
+
+        if let InvocableSysCall::Precise(sys_call) = name {
+            let Some(required_imports) = InvocableSysCall::required_imports_for_sys_call(sys_call) else {
+                return;
+            };
+
+            for &sys_call_import in required_imports {
+                self.enable_sys_call(InvocableSysCall::Loose(sys_call_import));
+            }
+        }
+    }
+
+    /// Enables given sys-call.
+    pub(crate) fn enable_sys_call(&mut self, name: InvocableSysCall) {
```

Comment:
```suggestion
    pub(crate) fn enable_sys_call_import(&mut self, name: InvocableSysCall) {
```

Reply:
done in 4a837cefad4e38ad9ac6e002d67e2bfac6e603da